frontend/model/contracts/shared/nativeNotification.js

```diff
@@ -16,6 +20,17 @@ export async function requestNotificationPermission (force: boolean = false): Pr
       return null
     }
   }
+
+  if (Notification.permission === 'granted' && notificationPayload) {
+    const isPushReady = await sbp('service-worker/check-push-subscription-ready')
+
+    if (isPushReady) {
+      sbp('service-worker/send-push', notificationPayload)
+    } else {
+      sbp('service-worker/setup-push-subscription', notificationPayload)
+    }
+  }
```

Comment:
A few concerns here:

1. We need to call `'service-worker/setup-push-subscription'` regardless of whether or not there's a `notificationPayload`, right? But this code doesn't do that.
2. The `notificationPayload` seems a bit out of place to me. The only reason you're using it is so that you can test the notifications in a sort of "debug" mode, but in the final version we're going to remove most of those notifications. It would be better to send them separately if permissions were granted, instead of here.